packages/flutter/lib/src/widgets/transitions.dart

```diff
@@ -1064,21 +1069,92 @@ class AnimatedBuilder extends AnimatedWidget {
-  /// depend on the animation, it's more efficient to build that subtree once
-  /// instead of rebuilding it on every animation tick.
+  /// {@macro flutter.widgets.transitions.AnimatedBuilder.optimizations}
+  final Widget? child;
+
+  @override
+  Widget build(BuildContext context) {
+    return builder(context, child);
+  }
+}
+
+/// A general-purpose widget for building a widget subtree when a [Listenable]
+/// changes.
+///
+/// [ListenableBuilder] is useful for more complex widgets that wish to listen
+/// to changes in other objects as part of a larger build function. To use
+/// [ListenableBuilder], simply construct the widget and pass it a builder
+/// function.
+///
+/// ## Performance optimizations
+///
+/// {@template flutter.widgets.transitions.ListenableBuilder.optimizations}
+/// If your [builder] function contains a subtree that does not depend on the
+/// [listenable], it's often more efficient to build that subtree once instead
+/// of rebuilding it on every change of the listenable.
+///
+/// If you pass a pre-built subtree as the [child] parameter, the
+/// [ListenableBuilder] will pass it back to your [builder] function so that you
+/// can incorporate it into your build.
+///
+/// Using this pre-built [child] is entirely optional, but can improve
+/// performance significantly in some cases and is therefore a good practice.
+/// {@endtemplate}
+///
+/// {@tool dartpad}
+/// This example shows how a [ListenableBuilder] can be used to listen to a
+/// [FocusNode] (which is also a [ChangeNotifier]) to see when a subtree has
+/// focus, and modify a decoration when its focus state changes.
+///
+/// ** See code in examples/api/lib/widgets/transitions/listenable_builder.0.dart **
+/// {@end-tool}
+///
+/// {@template flutter.flutter.ListenableBuilder.ChangeNotifier.rebuild}
+/// ## Improve rebuilds performance using [ListenableBuilder]
+///
+/// Any subtype of [Listenable] (such as [ChangeNotifier] or [ValueNotifier])
+/// can be used with a [ListenableBuilder] to rebuild only certain parts of a
+/// widget when the [Listenable] notifies its listeners. You can improve
+/// performance by specifying any widgets that don't need to change as a result
+/// of changes in the listener as the prebuilt [ListenableBuilder.child]
+/// attribute.
+///
+/// {@tool dartpad}
+/// The following example implements a simple counter that utilizes a
+/// [ListenableBuilder] to limit rebuilds to only the [Text] widget containing
+/// the count. The current count is stored in a [ValueNotifier], which rebuilds
+/// the [ListenableBuilder]'s contents when its value is changed.
+///
+/// ** See code in examples/api/lib/widgets/transitions/listenable_builder.1.dart **
+/// {@end-tool}
+/// {@endtemplate}
+///
+/// See also:
+///
+/// * [AnimatedBuilder], which has the same functionality, but is named more
+///   appropriately for a builder triggered by animations.
+class ListenableBuilder extends AnimatedWidget {
```

Comment:
Can you add some tests for this new widgets? Something to verify that builder is called when listenable notifies and maybe that the child is passed through as expected.

Reply:
Added a test for both `AnimatedBuilder` and `ListenableBuilder`, since there was never a test for `AnimatedBuilder`.